src/components/chat/chat_channels.jsx

```diff
@@ -51,7 +59,7 @@ const ChatChannels = React.createClass({
 
   getInitialState() {
     return {
-      currentChannel: ChatChannelsStore.getCurrentChannel(),
+      currentChannel: this.getParams().roomSlug,
```

Comment:
Sweet! React Router is new-ish to me, so just making sure I have a good handle on its ins and outs.